server.go

```diff
@@ -891,6 +893,170 @@ func (s *Server) HandleGetPayload(w http.ResponseWriter, r *http.Request) {
+		attribute.Bool("sszRequest", sszRequest),
+		attribute.Bool("sszResponse", sszResponse),
+		attribute.StringSlice("headers", headers),
+		attribute.String("slotUID", headerSlotUID),
+	)
+
+	bodyBytes, err := io.ReadAll(r.Body)
+	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
+		log.Error().Err(err).Msg("could not read registration")
+		respondError(getPayloadCtx, span, getPayloadV2, w, toErrorResp(http.StatusInternalServerError, "could not read registration"), &log, s.tracer)
+		return
+	}
+	signedBlindedBeaconBlock := new(common.VersionedSignedBlindedBeaconBlock)
+	if sszRequest {
+		_, decodeSSZSpan := s.tracer.Start(getPayloadCtx, "handleGetPayload-decodeSSZ")
+		err := signedBlindedBeaconBlock.UnmarshalSSZ(bodyBytes)
+		if err != nil {
+			log.Error().Err(err).Msg("failed to decode request payload")
+			decodeSSZSpan.End()
+			respondError(getPayloadCtx, span, getPayloadV2, w, toErrorResp(http.StatusInternalServerError, "failed to decode request payload"), &log, s.tracer)
+			return
+		}
+		decodeSSZSpan.End()
+		_, encodeJSONSpan := s.tracer.Start(getPayloadCtx, "handleGetPayload-encodeJSON")
+		bodyBytes, err = signedBlindedBeaconBlock.MarshalJSON()
+		if err != nil {
+			encodeJSONSpan.End()
+			log.Error().Err(err).Msg("failed to marshal to json")
+			respondError(getPayloadCtx, span, getPayloadV2, w, toErrorResp(http.StatusInternalServerError, "failed to marshal to json"), &log, s.tracer)
+			return
+		}
+		encodeJSONSpan.End()
+	}
+	span.AddEvent("handleGetPayload-svcGetPayload")
+	var (
+		versionedPayloadInfo *common.VersionedPayloadInfo
+	)
+	method := getPayloadV2
+	versionedPayloadInfo, err = s.svc.GetPayloadV2(getPayloadCtx, &log, &PayloadRequestParams{
+		ReceivedAt:                receivedAt,
+		Payload:                   bodyBytes,
+		ClientIP:                  clientIP,
+		AuthHeader:                authHeader,
+		ValidatorID:               validatorID,
+		AccountID:                 accountID,
+		GetPayloadStartTimeUnixMS: boostSendTime,
+		Cluster:                   cluster,
+		UserAgent:                 userAgent,
+		SlotUID:                   headerSlotUID,
+	})
+	_, mergeLogMetric := s.tracer.Start(getPayloadCtx, "handleGetPayload-mergeLogMetric")
+	if err != nil {
+		log.Error().Err(err).Msg("Error in GetPayload")
+		span.SetAttributes(
+			attribute.String("error", err.Error()),
+		)
+		span.SetStatus(codes.Error, err.Error())
+		respondError(getPayloadCtx, span, method, w, err, &log, s.tracer)
+		return
+	}
+	mergeLogMetric.End()
+
+	// Return response
+	if !sszResponse {
+		if err := respondOK(getPayloadCtx, span, method, w, versionedPayloadInfo.GetResponse(), &log, s.tracer); err == nil {
+			success = true
+		}
+		return
+	}
+	_, marshalUnmarshalSpan := s.tracer.Start(getPayloadCtx, "handleGetPayload-marshalUnmarshal")
+	payloadResponse := new(common.VersionedSubmitBlindedBlockResponse)
+	if err := payloadResponse.UnmarshalJSON(versionedPayloadInfo.GetResponse()); err != nil {
```

Comment:
Remove from 1039 - 1056. We are not gonna respond anything, just a success or failure